### PR TITLE
Partial crowdfund support

### PR DESCRIPTION
**Pull request by onggunhao**
- [x] Add principalDisbursed to terms contract
- [ ] Keep track of totalAmountRaised in `crowdLoan`
- [ ] startLoan in crowdloan should use totalAmountRaised as principalDisbursed
- [ ] rename interestRate to monthlyInterestRate for clarity

### CODE REVIEW

File: contracts/crowdloan/Crowdloan.sol [hunk 1]
```diff
@@ -25,6 +25,8 @@ contract Crowdloan is Initializable, ICrowdloan, ReentrancyGuard {
     ITermsContract public termsContract;
     IRepaymentManager public repaymentManager;
 
+    uint256 totalCrowdfunded;
```
Comment (adibas03) [inline contracts/crowdloan/Crowdloan.sol]: what is the purpose of `totalCrowdfunded`,
why do we have to update it on each fund, and not just pick the value once loan is changed from `funding` phase to `repayment` phase?

File: contracts/crowdloan/Crowdloan.sol [hunk 2]
```diff
@@ -142,31 +169,13 @@ contract Crowdloan is Initializable, ICrowdloan, ReentrancyGuard {
         );
 
         if (termsContract.getLoanStatus() < TermsContractLib.LoanStatus.REPAYMENT_CYCLE) {
-            termsContract.startLoan();
+            termsContract.startRepaymentCycle(totalCrowdfunded); // TODO(Dan): change this to be the amount actually raised
```
Comment (adibas03) [inline contracts/crowdloan/Crowdloan.sol]: this should be
`termsContract.startRepaymentCycle(_getPrincipalToken().balanceOf(address(this)))`,
which simply takes the balance at this point in time of the crowdloan contract.
This also removes the need to track `totalCrowdfunded` within the fund and refund functions.

Comment (onggunhao) [reply contracts/crowdloan/Crowdloan.sol]: @adibas03 Ah yes I thought about that. That approach might create a few problems:

1. Someone does a native ERC20 transfer to the smart contract, such that `_getPrincipalToken().balanceOf(address(this))` is > than `principalRequested`

2. In TermsContract, we have a `require` that checks that `principalDisbursed` is not more than `principalRequested`

I think I can resolve it by removing `totalCrowdfunded`, and instead just passing in the smaller of `balanceOf(address(this))` and `principalRequested` into the `startRepaymentCycle` function

File: contracts/crowdloan/Crowdloan.sol (hunk 1 above)
Comment (onggunhao) [reply contracts/crowdloan/Crowdloan.sol]: That makes sense - let me update this

File: contracts/crowdloan/Crowdloan.sol (hunk 2 above)
Comment (onggunhao) [reply contracts/crowdloan/Crowdloan.sol]: https://github.com/enabledao/enable-contracts/pull/51/commits/f1aba44064c75752a077121877d601b20bf37632

Comment (adibas03) [reply contracts/crowdloan/Crowdloan.sol]: If we are to use a ceiling, it means we have to implement a cap for the withdrawal function as well, which makes withdrawing in bits more annoying, and would be better we have a simple withdraw function that takes the lesser of `principalRequested` and `balanceOf(address(this))` and transfers the said amount, after which it locks withdrawal

Comment (onggunhao) [reply contracts/crowdloan/Crowdloan.sol]: Ah what I did was:

* No ceiling on withdrawals (uses `balanceOf(address(this))`)
* Terms Contract will only use a maximum of `principalRequested` to calculate interest and repayments

See https://github.com/enabledao/enable-contracts/commit/f1aba44064c75752a077121877d601b20bf37632

**Comment by adibas03** [comment]
@onggunhao Is this PR ready?

**Comment by adibas03** [comment]
moved to #54

**Pull request closed without merging**